src/python/pants/jvm/resolve/jvm_tool.py

```diff
@@ -276,6 +285,31 @@ async def gather_coordinates_for_jvm_lockfile(request: GatherJvmCoordinatesReque
     return Coordinates(coordinates)
 
 
+@rule
+async def load_jvm_lockfile(
+    request: JvmToolLockfileRequest,
+) -> CoursierResolvedLockfile:
```

Comment:
This `@rule` is the beginning of a cache-aware replacement for `JvmToolBase.resolved_lockfile`. To fully replace that method, it would need to support loading the default lockfile as a resource (...because `scalac_plugins` don't have one).